### PR TITLE
Feature/job progress

### PR DESCRIPTION
**Pull request by alexwelch**

### CODE REVIEW

File: app/models/job.rb [hunk 1]
```diff
@@ -1,10 +1,23 @@
 class Job < BaseResource
   has_one :override, class_name: 'JobOverride'
+  has_many :steps
 
   schema do
     integer :template_id
   end
 
+  def with_template!
```
Comment (alexwelch) [inline app/models/job.rb]: this can be removed if we include template_name in the response from the API

**Comment by alexwelch** [comment]
This PR was growing, so I opened a new, even longer PR!!!!!!! https://github.com/CenturyLinkLabs/panamax-ui/pull/500

**Comment by bdehamer** [comment]
:+1:

**Comment by dharmamike** [comment]
:+1:

**Pull request merged**